### PR TITLE
Let idempotent

### PR DESCRIPTION
**Pull request by Rizary**
there are changes in the test_data to match the output. Please bear in mind that we calculate the format based on the input.

### CODE REVIEW

File: src/rules.rs [hunk 1]
```diff
@@ -105,9 +105,11 @@ pub(crate) fn spacing() -> SpacingDsl {
...
         .inside(NODE_LET_IN).after(T![let]).single_space_or_optional_newline()
         .inside(NODE_LET_IN).around(T![in]).single_space_or_optional_newline()
-        .inside(NODE_LET_IN).before(T![in]).when(header_has_multi_value).when(before_token_not_newline).newline()
-        .inside(NODE_LET_IN).after(T![in]).when(in_body_newline).newline()
-        .inside(NODE_LET_IN).before(NODE_KEY_VALUE).when(header_has_multi_value).when(before_token_not_newline).newline()
+        .inside(NODE_LET_IN).after(NODE_KEY_VALUE).single_space_or_optional_newline()
+        .inside(NODE_LET_IN).before(NODE_KEY_VALUE).when(let_header_has_newline).newline()
+        // .inside(NODE_LET_IN).before(T![in]).when(header_has_multi_value).when(before_token_not_newline).newline()
+        .inside(NODE_LET_IN).around(T![in]).when(let_header_has_newline).newline()
+        // .inside(NODE_LET_IN).before(NODE_KEY_VALUE).when(header_has_multi_value).when(before_token_not_newline).newline()
```
Comment (zimbatm) [inline src/rules.rs:112]: can you please remove the commented code?

File: src/rules.rs [hunk 2]
```diff
@@ -150,11 +153,11 @@ pub(crate) fn spacing() -> SpacingDsl {
...
-        .add_rule(dsl::SpacingRule {
-            name: None,
-            pattern: p(T![let]) & p(after_let_newline),
-            space: dsl::Space { loc: dsl::SpaceLoc::After, value: dsl::SpaceValue::Newline }
-        })
+        // .add_rule(dsl::SpacingRule {
+        //     name: None,
+        //     pattern: p(T![let]) & p(after_let_newline),
+        //     space: dsl::Space { loc: dsl::SpaceLoc::After, value: dsl::SpaceValue::Newline }
+        // })
```
Comment (zimbatm) [inline src/rules.rs:160]: same

File: test_data/issue-125.good.nix [hunk 3]
```diff
@@ -15,8 +15,10 @@ let
     in
     x;
   baz =
-    let x = 3;
-    in x;
```
Comment (zimbatm) [inline test_data/issue-125.good.nix:19]: This is valid syntax. I don't think we should break that.

File: test_data/issue-152.good.nix [hunk 4]
```diff
@@ -7,7 +7,8 @@ let
     #comment
     x + x;
   bar = x:
-    let y = x;
```
Comment (zimbatm) [inline test_data/issue-152.good.nix:10]: also valid

File: test_data/curried_fn.good.nix [hunk 5]
```diff
@@ -1,6 +1,8 @@
 {
   mkValueStringDefault = {}: v: with builtins;
-    let err = t: v: abort
-      "oups";
-    in 92;
+    let
+      err = t: v: abort
+        "oups";
+    in
+    92;
```
Comment (zimbatm) [inline test_data/curried_fn.good.nix:7]: why is 92 pushed to the next line?

File: test_data/issue-125.good.nix (hunk 3 above)
Comment (Rizary) [reply test_data/issue-125.good.nix:19]: because it contains newline between `let` and `in`

File: test_data/curried_fn.good.nix (hunk 5 above)
Comment (Rizary) [reply test_data/curried_fn.good.nix:7]: because it contains newline between `let` and `in`

File: src/rules.rs (hunk 1 above)
Comment (Rizary) [reply src/rules.rs:112]: I'll clean it up.

File: test_data/issue-125.good.nix (hunk 3 above)
Comment (zimbatm) [reply test_data/issue-125.good.nix:19]: There should be an exception if the new line is between the `in` and the previous token.

**Review by zimbatm — APPROVED** [review]
Looks good, thansk!

**Pull request merged**